src/compiler/moduleNameResolver.ts

```diff
@@ -1845,7 +1849,9 @@ namespace ts {
             if (typeof existing !== "boolean") {
                 if (traceEnabled) trace(host, Diagnostics.File_0_exists_according_to_earlier_cached_lookups, packageJsonPath);
                 state.affectingLocations.push(packageJsonPath);
-                return existing;
+                return existing.packageDirectory === packageDirectory ?

```

Comment:
This is a way to reuse the cache contents with a variant (e.g. casing, slashes) of the path?